src/com/goide/inspections/GoPlaceholderChecker.java

```diff
@@ -0,0 +1,502 @@
+/*
+ * Copyright 2013-2016 Sergey Ignatov, Alexander Zolotov, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.inspections;
+
+import com.goide.psi.*;
+import com.goide.psi.impl.GoTypeUtil;
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.util.Couple;
+import com.intellij.openapi.util.Pair;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiReference;
+import com.intellij.util.containers.ContainerUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static com.goide.GoConstants.TESTING_PATH;
+import static com.goide.inspections.GoPlaceholderChecker.PrintfArgType.*;
+
+public class GoPlaceholderChecker {
+
+  // This holds the name of the known formatting functions and position of the string to be formatted
+  public static final Map<String, Integer> FORMATTING_FUNCTIONS = ContainerUtil.newHashMap(
+    Pair.pair("errorf", 0),
+    Pair.pair("fatalf", 0),
+    Pair.pair("fprintf", 1),
+    Pair.pair("fscanf", 1),
+    Pair.pair("logf", 0),
+    Pair.pair("panicf", 0),
+    Pair.pair("printf", 0),
+    Pair.pair("scanf", 0),
+    Pair.pair("skipf", 0),
+    Pair.pair("sprintf", 0),
+    Pair.pair("sscanf", 1));
+
+  public static final List<String> PRINT_FUNCTIONS = ContainerUtil.newArrayList(
```

Comment:
1. the field is used only in `GoPlaceholderCountInspection`, so why it's here and public?
2. if the list is used for `contains` operation only it shouldn't be a list.